src/main/java/de/tum/in/www1/artemis/web/rest/plagiarism/PlagiarismResource.java

```diff
@@ -0,0 +1,114 @@
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.plagiarism.PlagiarismService;
+import de.tum.in.www1.artemis.web.rest.dto.PlagiarismComparisonStatusDTO;
+import de.tum.in.www1.artemis.web.rest.errors.AccessForbiddenException;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+
+/**
+ * REST controller for managing TextExercise.
+ */
+@RestController
+@RequestMapping("api/")
+public class PlagiarismResource {
+
+    private final CourseRepository courseRepository;
+
+    private final AuthorizationCheckService authenticationCheckService;
+
+    private final UserRepository userRepository;
+
+    private final Logger log = LoggerFactory.getLogger(PlagiarismResource.class);
+
+    private final PlagiarismComparisonRepository plagiarismComparisonRepository;
+
+    private final PlagiarismService plagiarismService;
+
+    public PlagiarismResource(PlagiarismComparisonRepository plagiarismComparisonRepository, CourseRepository courseRepository,
+            AuthorizationCheckService authenticationCheckService, UserRepository userRepository, PlagiarismService plagiarismService) {
+        this.plagiarismComparisonRepository = plagiarismComparisonRepository;
+        this.courseRepository = courseRepository;
+        this.authenticationCheckService = authenticationCheckService;
+        this.userRepository = userRepository;
+        this.plagiarismService = plagiarismService;
+    }
+
+    /**
+     * Update the status of the plagiarism comparison with the given ID.
+     * I.e. An editor or instructor sees a possible plagiarism case for the first time and decides if it should be further examined, or if it is not a plagiarism.
+     *
+     * @param comparisonId of the plagiarism comparison to update the status of
+     * @param courseId the id of the course
+     * @param statusDTO new status for the given comparison
+     * @return the ResponseEntity with status 200 (Ok) or with status 400 (Bad Request) if the parameters are invalid
+     */
+    @PutMapping("courses/{courseId}/plagiarism-comparisons/{comparisonId}/status")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<Void> updatePlagiarismComparisonStatus(@PathVariable("courseId") long courseId, @PathVariable("comparisonId") long comparisonId,
+            @RequestBody PlagiarismComparisonStatusDTO statusDTO) {
+        log.info("REST request to update the status {} of the plagiarism comparison with id: {}", statusDTO.getStatus(), comparisonId);
+        Course course = courseRepository.findByIdElseThrow(courseId);
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        authenticationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, course, user);
+
+        // TODO: this check can take up to a few seconds in the worst case, we should do it directly in the database
```

Comment:
What check are you talking about here? I don't see anything that applies to this comment. Or do you mean the `plagiarismService.updatePlagiarismComparisonStatus` call below? If that is the case you should probably move the comment down.

Reply:
It's actually this check from the plagiarism comparison repository: `findByIdWithSubmissionsStudentsElseThrow`. This is due to the implementation of the plagiarism submissions and their elements. This comment was already there and was added when it caused some servers to crash (it is fixed now but can still be improved), so I just left it in as I didn't update the functionality of this method.